File: app/src/main/java/com/mk/room/db/sample/RoomDB/MainDao.java
package com.mk.room.db.sample.RoomDB;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface MainDao {
    @Insert(onConflict =  REPLACE)
    void insert(MainData mainData);

    @Delete
    void delete(MainData mainData);

    @Delete
    void deleteAll(List<MainData> mainData);

    @Query("UPDATE table_name SET text = :txt WHERE ID = :ID")
    void update(int ID , String txt);

    @Query("SELECT * FROM table_name")
    List<MainData> getAll();
}

File: app/src/main/java/com/mk/room/db/sample/RoomWithKotlin/MainDao.kt
package com.mk.room.db.sample.RoomWithKotlin

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface  MainDao {
    @Query("SELECT * FROM tableName")
    fun getAll(): List<MainData>

    @Query("UPDATE tableName SET input_text = :inputText WHERE ID = :ID")
    fun update(ID: Int,inputText: String)

    @Insert
    fun insertAll(mainData: MainData)
    @Delete
    fun delete(mainData: MainData)
    @Delete
    fun deleteAll(mainData: List<MainData>)


}